src/workerd/util/sqlite.c++

```diff
@@ -784,8 +792,12 @@ void SqliteDatabase::Query::bind(uint i, decltype(nullptr)) {
 }
 
 void SqliteDatabase::Query::nextRow() {
+  KJ_ASSERT(db.currentStatement == nullptr, "recursive nextRow()?");
+  KJ_DEFER(db.currentStatement = nullptr);
+  db.currentStatement = *statement;
```

Comment:
Interesting, I'm a little surprised there's just one entrypoint where we need to track this. Very convenient.